### PR TITLE
Add support for LoggingLevelSwitch

### LINKED ISSUE #69: LoggingLevelSwitch support
See: https://github.com/serilog/serilog-settings-appsettings/issues/7

### PR DESCRIPTION
**Pull request by tsimbalar**
fixes #69 

Supports : 
- declaration of LoggingLevelSwitch 
- MinimumLevel.ControlledBy(switch)
- MinimumLevel.Override(prefix, switch)
- passing a switch by reference as the argument to a Sink (or any callable method)

JSON syntax : 
```json
{
  "Serilog": {
    "Using": [ "TestDummies" ],
    "LevelSwitches": { "$mySwitch": "Warning" },
    "MinimumLevel": {
      "ControlledBy": "$mySwitch"
    },
    "WriteTo": [
      {
        "Name": "DummyWithLevelSwitch",
        "Args": {
          "controlLevelSwitch": "$mySwitch"
        }
      }
    ]
  }
}
```

but also overrides : 

```json
{
  "Serilog": {
    "Using": [ "TestDummies" ],
    "LevelSwitches": { "$mySwitch": "Warning" },
    "MinimumLevel": {
      "Default": "Information",
      "Override": {
        "Microsoft": "$mySwitch"
      }
    },
    "WriteTo": [
      {
        "Name": "DummyWithLevelSwitch",
        "Args": {
          "controlLevelSwitch": "$mySwitch"
        }
      }
    ]
  }
}
```

In order to implement "look up a switch by name", I had to change `IConfigurationArgumentValue`'s method `object ConvertTo(Type toType)` to `object ConvertTo(Type toType, IReadOnlyDictionary<string, LoggingLevelSwitch> declaredLevelSwitches)` and cascade that change everywhere ...

### CODE REVIEW

File: src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs [hunk 1]
```diff
@@ -37,14 +40,48 @@ public ConfigurationReader(IConfigurationSection configuration, DependencyContex
...
-        void ApplyMinimumLevel(LoggerConfiguration loggerConfiguration)
+            var declaredLevelSwitches = ProcessLevelSwitchDeclarations();
+            ApplyMinimumLevel(loggerConfiguration, declaredLevelSwitches);
+            ApplyEnrichment(loggerConfiguration, declaredLevelSwitches);
+            ApplyFilters(loggerConfiguration, declaredLevelSwitches);
+            ApplySinks(loggerConfiguration, declaredLevelSwitches);
+            ApplyAuditSinks(loggerConfiguration, declaredLevelSwitches);
+        }
+
+        private IReadOnlyDictionary<string, LoggingLevelSwitch> ProcessLevelSwitchDeclarations()
```
Comment (nblumhardt) [inline src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs]: Very minor nitpick- the Serilog repos don't use explicit `private` or `internal` (I think there might be a DotSettings file in the main repo.)

**Comment by nblumhardt** [comment]
Looks awesome 👍 

Just added one minor nitpick - keen to get this to `dev` so we can kick the tires! :-)

File: src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs (hunk 1 above)
Comment (tsimbalar) [reply src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs]: yes indeed :) I'll copy over the `.DotSettings` file from the main repo to detect those cases.

**Comment by tsimbalar** [comment]
I've updated the code based on your comments :)

**Comment by nblumhardt** [comment]
🎉

**Pull request merged**